designer/server/src/test/scala/pl/touk/nussknacker/ui/integration/DictsFlowTest.scala

```diff
@@ -114,6 +117,164 @@ class DictsFlowTest
       .rightValue shouldBe """{"key":"true","label":"ON"}""" // returns "ON" even though "OLD LABEL" was sent, because of ProcessDictSubstitutor
   }
 
+  test("handle labels correctly even if referenced fragment compilation fails - fragment with invalid end of branch") {
+    val fragmentName = "test_frag"
+
+    createEmptyScenario(ProcessName(fragmentName), isFragment = true)
+
+    // this fragment has only an input node, no valid end of branch
+    val fragmentJson = """{
+                         |  "scenarioGraph": {
+                         |    "properties": {
+                         |      "additionalFields": {
+                         |        "description": "desc",
+                         |        "properties": {},
+                         |        "metaDataType": "FragmentSpecificData"
+                         |      }
+                         |    },
+                         |    "nodes": [
+                         |      {
+                         |        "parameters": [
+                         |          {
+                         |            "uuid": "98de9a8c-4e2c-44ae-8e5a-1136b5fffbec",
+                         |            "name": "test",
+                         |            "required": false,
+                         |            "valueEditor": {
+                         |              "allowOtherValue": false,
+                         |              "dictId": "rgb",
+                         |              "type": "ValueInputWithDictEditor"
+                         |            },
+                         |            "typ": {
+                         |              "refClazzName": "java.lang.String"
+                         |            }
+                         |          }
+                         |        ],
+                         |        "type": "FragmentInputDefinition",
+                         |        "branchParametersTemplate": [],
+                         |        "id": "input"
+                         |      }
+                         |    ],
+                         |    "edges": []
+                         |  },
+                         |  "comment": ""
+                         |}""".stripMargin
+
+    val globalErrorFrag = extractGlobalValidationResult(fragmentName, fragmentJson)
+    globalErrorFrag.map(_.error.typ) shouldBe List("InvalidTailOfBranch")
```

Comment:
This error type looks hacky in this context. I'd rather expect sth like usage of/reference to invalid fragment . Is it easy to change? Can it be node error instead of global error?